### PR TITLE
app-emulation/libvirt: Backport fixes for two CVEs

### PR DESCRIPTION
**Pull request by zippy2**
Two CVEs were identified recently. Unfortunately, they affect nearly all versions in portage. On the up side - both flaws can be triggered only when using libvirt's RPC directly, i.e. apps using libvirt APIs are NOT affected.

### CODE REVIEW

File: app-emulation/libvirt/libvirt-10.1.0.ebuild [hunk 1]
```diff
@@ -147,6 +147,7 @@ PATCHES=(
 	"${FILESDIR}"/${PN}-9.4.0-fix_paths_in_libvirt-guests_sh.patch
 	"${FILESDIR}"/${PN}-9.9.0-do-not-use-sysconfig.patch
 	"${FILESDIR}"/${PN}-9.6.0-fix-paths-for-apparmor.patch
+	"${FILESDIR}"/${PN}-10.2.0-remote-check-for-negative-array-lengths-before-alloc.patch
```
Comment (thesamesam) [inline app-emulation/libvirt/libvirt-10.1.0.ebuild:150]: Revbumps are required for runtime changes.

Comment (zippy2) [reply app-emulation/libvirt/libvirt-10.1.0.ebuild:150]: Let me fix that.

**Comment by thesamesam** [comment]
Sorry I didn't notice this before -- would you mind filing a security bug on bugs.gentoo.org and tagging it with `Bug:` in this commit?

Also, note that while `git mv` is not ideal for patches sometimes, I think it's okay here, as the change is obviously right

**Comment by zippy2** [comment]
> Sorry I didn't notice this before -- would you mind filing a security bug on bugs.gentoo.org and tagging it with `Bug:` in this commit?

Sure thing.

> Also, note that while `git mv` is not ideal for patches sometimes, I think it's okay here, as the change is obviously right

How come? What else should I used?

**Comment by thesamesam** [comment]
The reason being that patches may introduce regressions, and by doing `git mv` here, you're not allowing people to downgrade easily and we also circumvent the stabilisation process.

But in this case, they're just introducing bounds checks, so given it's a security bug, it's probably fine.

The alternative is to `cp` each relevant version to -rN+1 and `ekeyword ~all` them, then file a stable bug (shorter than 30 days like usual) for them quickly.

**Comment by zippy2** [comment]
Ah, nuances of upstream vs downstream development.

**Pull request closed without merging**